mdd.py

```diff
@@ -346,35 +347,73 @@ def get_graphics_info():
 
-        # Run xrandr command and capture output
-        xrandr_output = get_command_output("xrandr")
-        if xrandr_output:
-            outputs = []
-            output_connected = False
-
-            for line in xrandr_output.split("\n"):
-                connected_match = re.match(r"^(\S+) connected", line)
-                if connected_match:
-                    output_connected = True
-                    continue
-
-                if output_connected:
-                    mode_match = re.match(r"^   (\d+x\d+)\s+([\d.]+)\*", line)
-                    if mode_match:
-                        resolution = mode_match.group(1)
-                        try:
-                            refresh = float(mode_match.group(2))
-                        except ValueError:
-                            refresh = 0
-                        outputs.append(
-                            {
-                                "model": "unknown",
-                                "res": resolution,
-                                "refresh": refresh,
-                                "dpi": None,
-                                "size": None,
-                            }
-                        )
+    # Run xrandr command and capture output
+    if xrandr_output := get_command_output("xrandr"):
+        mapped = None
+
+        for line in xrandr_output.split("\n"):
+            if re.match(r"^(\S+) disconnected", line):
+                mapped = None
+                continue
+
+            if re.match(r"^(\S+) connected", line):
+                # When we encounter a line that contains the word 'connected', we mark the beginning
+                # of a new connected output and continue with the line afterwards.
+                mapped = line.split(" ")
+                found_output = found_output = next(
+                    (output for output in outputs if output["mapped"] == mapped[0]),
+                    None,
+                )
+
+                continue
+
+            mode_match = re.match(r"^   (\d+x\d+)\s+([\d.]+)\*", line)
+            if not mode_match:
+                # Inactive mode
+                continue
+
+            if not mapped:
+                # A mode can only be active for mapped outputs
+                raise Exception(f"matched a mode '{mode_match}' without being mapped")
+
+            resolution = mode_match.group(1)
+            try:
+                refresh = float(mode_match.group(2))
+            except ValueError:
+                refresh = None
+
+            # Try to get size from connected line.
+            size = ""
+            if len(mapped) > 2:
+                try:
+                    height = int(mapped[-1].replace("mm", ""))
+                    width = int(mapped[-3].replace("mm", ""))
+                    size = str(height) + "x" + str(width)
+                except:
+                    logging.error(f"xrandr size: {mapped}")
+                    pass
+
+            if found_output:
```

Comment:
where does the condition originate?

you are using it in line 404 - but where does it come from ?

Reply:
I found it in line 363 hidden in a if condition

I would initialize it to False - then change value as needed